### PR TITLE
Parallel extract merge

### PR DESCRIPTION
**Pull request by macintux**
Incorporate changes to support 2i parallel extract and return_body.

### CODE REVIEW

File: src/riakc_pb_socket.erl [hunk 1]
```diff
@@ -1872,24 +1948,38 @@ process_response(#request{msg = #tsqueryreq{}},
...
     ?INDEX_STREAM_RESULT{keys=Keys};
-process_index_response(true, [], Results) ->
-    %% rpbpair is abused to send Value,Key pairs as Key, Value pairs
-    %% in a 2i query the 'key' is the index value and the 'value'
-    %% the indexed objects primary key
+process_index_response(_, [], Results) ->
+    %% If return_terms is true and return_body is false, rpbpair is
+    %% abused to send Value,Key pairs as Key, Value pairs in a 2i
+    %% query the 'key' is the index value and the 'value' the indexed
+    %% objects primary key.
```
Comment (mrallen1) [inline src/riakc_pb_socket.erl]: This comment does not apply any more. Please clean up.

**Comment by mrallen1** [comment]
Ready to go once the comment is cleaned up

**Comment by mrallen1** [comment]
:+1: 0c9c65c

**Comment by macintux** [comment]
@borshop merge

**Pull request merged**